components/servo/lib.rs

```diff
@@ -1091,6 +1098,9 @@ fn default_user_agent_string_for(agent: UserAgent) -> &'static str {
     const DESKTOP_UA_STRING: &'static str =
         "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Servo/1.0 Firefox/111.0";
 
+    #[cfg(target_os = "android")]
```

Comment:
It might be a good moment to clean this up and just conditionally define a `UA_STRING` and always return it.

Reply:
Hmm, the existing code is written so that the user can switch user agent to ios/android specific ones even on desktop at runtime using [this](https://github.com/servo/servo/blob/main/ports/servoshell/main2.rs#L38) flag, probably to allow testing.

I think implementing what you suggested would require removing this functionality from libservo (maybe we can do it to ports/servoshell). Can this be tackled in a separate PR?

Reply:
Yes, sure!